server/src/metadata/field-metadata/field-metadata.entity.ts

```diff
@@ -11,29 +11,32 @@ import {
 } from 'typeorm';
 import {
   Authorize,
+  BeforeCreateOne,
   IDField,
   QueryOptions,
 } from '@ptc-org/nestjs-query-graphql';
 
 import { ObjectMetadata } from 'src/metadata/object-metadata/object-metadata.entity';
 
+import { BeforeCreateOneField } from './hooks/before-create-one-field.hook';
+
 export type FieldMetadataTargetColumnMap = {
   [key: string]: string;
 };
-
+@Entity('field_metadata')
 @ObjectType('field')
+@BeforeCreateOne(BeforeCreateOneField)
```

Comment:
This hook is called before createOneField function inside the service. Again, I'm not really a big fan of all those annotations since it makes the flow very hard to understand if you don't know the package, we'll see if we can improve on that later.

Reply:
@Weiko Not sure if it's the best way to do it, I'll try to take a look